### PR TITLE
Add some helpers for testing to its own package in app

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Adds some helpers for testing various functions in a re-usable way in app.go

I think this should probably be paired with more custom functions / integrations in modules, to get a more composable testing framework. This at least does a first step with setting up validators, and epochs across modules.

I am torn on whether it should be suiteI, or an embeddable struct though. I'm leaning towards making it an embeddable struct.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Review by mconcat — APPROVED** [review]
I think this is overall great PR that encapsulates common logics to SuiteI!

**Review by mattverse — APPROVED** [review]
This is awesome! We should also aim to write more integration tests using these, as imo they provide a "flow test". 
 
Also, I'm planning to write up a follow up issue on doing the same thing for commonly used methods such as `LockTokens` if you agree?

File: app/apptesting/test_suite.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+		time.Unix(0, 0),
+		false,
+		0,
+	)
+	suite.GetApp().SlashingKeeper.SetValidatorSigningInfo(suite.GetCtx(), consAddr, signingInfo)
+
+	return valAddr
+}
+
+func BeginNewBlock(suite SuiteI, executeNextEpoch bool) {
```
[the remark this answers is not in the document]
Comment (ValarDragon) [reply app/apptesting/test_suite.go:60]: yeah agreed. Would need to make suite a struct to make it a method (AFAICT, you can't do methods on an interface sadly)

**Comment by ValarDragon** [comment]
Any thoughts on struct vs interface? I think interface gives us better composability / more re-usability across codebases, but struct will simplify the code a lot.

I'm leaning towards struct, since these whole keeper test suites are kinda odd in that they depend on the OsmosisApp already. I'm generally not a big fan of how they depend on the entirety of the app, but its alright

**Comment by mattverse** [comment]
@ValarDragon can I go ahead and merge this?

**Pull request merged**